datafusion/physical-expr/src/aggregate/count_distinct.rs

```diff
@@ -205,6 +146,52 @@ impl Accumulator for DistinctCountAccumulator {
 
         Ok(cols_out.into_iter().collect())
     }
+    fn update_batch(&mut self, values: &[ArrayRef]) -> Result<()> {
+        if values.is_empty() {
+            return Ok(());
+        }
+        let arr = &values[0];
+        (0..arr.len()).try_for_each(|index| {
+            let scalar = ScalarValue::try_from_array(arr, index)?;
+            let scalar = vec![scalar];
```

Comment:
We can avoid creating a `Vec` here

Reply:
Hello @Dandandan, thank you for reviewing my work. I'm currently having difficulty finding a way to avoid using `vec` in this context. Could you please provide me with some guidance on how to refactor it?

Reply:
We can modify `DistinctScalarValues` from
`struct DistinctScalarValues(Vec<ScalarValue>)`
to 
`struct DistinctScalarValues(ScalarValue)`

and update the code accordingly.

This might provide some small performance increase.